server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -478,8 +478,14 @@ private void init()
           jp = objectMapper.getFactory().createParser(future.get());
           final JsonToken nextToken = jp.nextToken();
           if (nextToken == JsonToken.START_OBJECT) {
-            QueryInterruptedException e = jp.getCodec().readValue(jp, QueryInterruptedException.class);
-            throw e;
+            QueryInterruptedException cause = jp.getCodec().readValue(jp, QueryInterruptedException.class);
+            //case we get an exception with an unknown message.
+            if (cause.isNotKnown()) {
+              throw new QueryInterruptedException(QueryInterruptedException.UNKNOWN_EXCEPTION, cause.getMessage(), host);
+            } else {
+              throw  new QueryInterruptedException(cause, host);
```

Comment:
Given that this is deserializing a remote exception, this "re-wrapping" works as a demarcation point between the remote and the local.  Seems potentially useful to me...  If nothing else, it might just be some extra stuff on the exception, but it shouldn't make it more difficult to track down what happened.